workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -542,7 +542,7 @@ namespace <#= qualifiedNamespace #>
                     dirtyEvents += events<#= eventDetail.EventName #>.Count;
 <# } #>
 
-                    if (data.DirtyBit || dirtyEvents > 0)
+                    if (data.IsDirty() || dirtyEvents > 0)
                     {
                         var update = new global::Improbable.Worker.Core.SchemaComponentUpdate(<#= componentDetails.ComponentId #>);
                         <#= componentNamespace #>.Serialization.SerializeUpdate(data, update);
```

Comment:
This can now be skipped if `!data.IsDirty()` It'll save a lot of checking on components that just had events updated